dev/src/lobster/typecheck.h

```diff
@@ -313,6 +330,20 @@ struct TypeChecker {
         return a == b;  // Not inlined for documentation purposes.
     }
 
+    bool TypeOfUndefinedGenericSpec(TypeRef t) {
+        if (!IsUDT(t->t)) return false;
+        return IsUndefinedGenericSpec(t->udt);
+    }
+
+    bool IsUndefinedGenericSpec(UDT* udt) {
+        for (int i = 0; i < udt->generics.size(); i++) {
+            auto& utr = udt->generics[i].giventype.utr;
+            if (!utr.Null() && utr == type_undefined)
```

Comment:
when can these be null?

Reply:
hmm I used to have a this check https://github.com/aardappel/lobster/blob/4bb2e43add47955916c16a5f9b914ab08cb783d3/dev/src/lobster/typecheck.h#L711 in different place, but you're right now null isn't needed; removed